boards/arm/nrf51_ble400/nrf51_ble400.dts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2018 Roman Tataurov <diytronic@yandex.ru>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf51822.dtsi>
+
+/ {
+	model = "Waveshare BLE400";
+	compatible = "waveshare,BLE400", "nordic,nrf51822-qfaa", "nordic,nrf51822";
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+};
+
```

Comment:
If you want to enable firmware update on this board you can add a flash partition layout. Not mandatory, just pointing it out in case you want to do it.

Reply:
Yes - saw it on other boards, but did not tested it. This is the only reason why it absent here.